sw/device/examples/hello_world/BUILD

```diff
@@ -0,0 +1,47 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+load("//rules:opentitan.bzl", "OPENTITAN_CPU", "opentitan_binary")
+
+opentitan_binary(
+    name = "hello_world",
+    srcs = [
+        "hello_world.c",
+    ],
+    copts = [
+        "-nostdlib",
+        "-ffreestanding",
+        # Disable the date-time warning only for the hello world program.
+        "-Wno-date-time",
+    ],
+    linkopts = [
+        "-nostartfiles",
+        "-nostdlib",
+    ],
+    deps = [
+        ":hello_world_lib",
+        "//sw/device/lib/base:mmio",
+    ],
+)
+
+cc_library(
+    name = "hello_world_lib",
+    target_compatible_with = [OPENTITAN_CPU],
+    deps = [
+        "//hw/top_earlgrey/sw/autogen:top_earlgrey",
+        "//sw/device/examples:demos",
+        "//sw/device/exts/common",
+        "//sw/device/lib:irq",
+        "//sw/device/lib:pinmux",
+        "//sw/device/lib/arch:device",
+        "//sw/device/lib/crt",
```

Comment:
The CRT init is kind of "special". Maybe let's leave it for now, but I think it needs to be bundled as part of the execution target, rather than being an ordinary dependency. The same goes for the linker scripts.

Reply:
Agreed, but I think we should refactor the CRT target definitions once we have build rules in place for the different kinds of things we want to link (boot_rom, rom_ext, examples, etc).

Reply:
Sounds good.